core/src/test/java/eu/fasten/core/maven/data/DependencyTest.java

```diff
@@ -15,25 +15,285 @@
 
 import java.util.HashMap;
 import java.util.HashSet;
 import java.util.Set;
+import java.util.function.Consumer;
 
 import org.json.JSONArray;
 import org.json.JSONObject;
 import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.Test;
 
+@SuppressWarnings("deprecation")
 public class DependencyTest {
 
-    private static final Dependency SOME_DEPENDENCY = new Dependency("gid", "aid",
-            Set.of(new VersionConstraint("(1.2.3,2.3.4]")), Set.of(new Exclusion("gid2", "aid2")), "compile", false,
-            "jar", "sources");
+    @Test
+    public void defaults() {
+        var sut = new Dependency();
+        assertEquals(0, sut.id);
+        assertEquals("", sut.getGroupId());
+        assertEquals("", sut.getArtifactId());
+
+        assertEquals("", sut.getVersion());
+        assertEquals(Set.of(), sut.getVersionConstraints());
+        assertArrayEquals(new String[0], sut.getVersionConstraintsArr());
+
+        assertEquals(Set.of(), sut.getExclusions());
+        assertEquals("", sut.getClassifier());
+        assertEquals("jar", sut.getPackagingType());
+        assertEquals(COMPILE, sut.getScope());
+
+        assertFalse(sut.isOptional());
+    }
+
+    @Test
+    public void nonDefaults() {
+        var sut = getSomeDep();
+        assertEquals(0, sut.id);
+        assertEquals("gid", sut.getGroupId());
+        assertEquals("aid", sut.getArtifactId());
+
+        assertEquals("(1.2.3,2.3.4]", sut.getVersion());
+        assertEquals(Set.of(VersionConstraint.init("(1.2.3,2.3.4]")), sut.getVersionConstraints());
+        String[] x = sut.getVersionConstraintsArr();
+        assertArrayEquals(new String[] { "(1.2.3,2.3.4]" }, x);
+
+        assertEquals(Set.of(Exclusion.init("gid2", "aid2")), sut.getExclusions());
+        assertEquals("sources", sut.getClassifier());
+        assertEquals("pom", sut.getPackagingType());
+        assertEquals(PROVIDED, sut.getScope());
+
+        assertTrue(sut.isOptional());
+    }
+
+    @Test
+    public void hasToString() {
+        var actual = new Dependency().toString();
+        assertTrue(actual.contains(Dependency.class.getSimpleName()));
+        assertTrue(actual.contains("@"));
+        assertTrue(actual.contains("\n"));
+        assertTrue(actual.contains("versionConstraints"));
+    }
+
+    @Test
+    public void toGA() {
+        var actual = new Dependency("g", "a", "1.2.3").toGA();
+        var expected = "g:a";
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void equalityDefaults() {
+        var a = new Dependency();
+        var b = new Dependency();
+        assertEquals(a, b);
+        assertEquals(a.hashCode(), b.hashCode());
+    }
+
+    @Test
+    public void equalityNonDefaults() {
+        var a = getSomeDep();
+        var b = getSomeDep();
+        assertEquals(a, b);
+        assertEquals(a.hashCode(), b.hashCode());
+    }
+
+    @Test
+    public void equalitDiffVersionsSet() {
```

Comment:
Maybe a typo in function names from here onwards, equalit -> equality?

Reply:
yes, fixed.